### ISSUE TITLE
Removal of outdated records is not correct

### ISSUE THREAD

**Issue by mikeubell** [caciviclab/disclosure-backend#252]
The code in xformnetfilerawdata.py starting at line 573 cannot be correct.  I depends on the date that the data actually processed.  Records are deleted if the current date is before or after the range of the reporting period.  If one were to run the data load near midnight two records from the same form could be handled differently if one was processed before midnight and one was processed after midnight.
Records from a 497 can be deleted when a subsequent 460 is processed.  It has nothing to do with when the code is actually run.  The fact that the reporting period has passed does not mean that the data has been filed.  Someone could file the data late or not at all.

**Comment by KyleW** [comment]
I agree with Mike that we need some more thought here. There needs to be an interim solution, however there might be a chance for us to do something cool here. 

Maybe, we come up with a way to score similarity between new records and old records and have a manual review for anything that looks reasonably close. Invite folks to tag some records as the same or not the same. It could be a cool way to get people involved in the process.

**Comment by mikeubell** [comment]
I chatted with David Montgomery @netifle.com.  He agrees that using the current time does not belong in duplicate processing. He said that a reasonable approach is to just check for duplicates on insert.  The transaction ID is unique within a filing and for those using netfile software to do the filings the transaction ID will be the same between a 496/7 transaction and the overriding 460 transaction.  I propose that we use the same technique that we did in opendisclosure.io which is to just use first_or_create using the date, amount, benefactor and beneficiory.  We did use the transaction id in addition and we could do that if we believe that users will use the netfile filing software or something that behaves the same way.
